### PR TITLE
feat: build time version flag

### PR DESCRIPTION
**Pull request by dunqan**
Introduce build time feature level flag, to accomodate breaking changes

Closes [HRZ-89610](https://spryker.atlassian.net/browse/HRZ-89610)

[HRZ-89610]: https://spryker.atlassian.net/browse/HRZ-89610?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: libs/platform/experience/src/routes.ts [hunk 1]
```diff
@@ -52,14 +50,19 @@ export const defaultExperienceRoutes: RouteConfig[] = [
...
   },
-  {
-    path: '/product/:sku',
-    type: RouteType.Product,
-  },
-  {
-    path: '/category/:id',
-    type: RouteType.Category,
-  },
+  ...(OryxFeatureLevel < 110
```
Comment (tolerants) [inline libs/platform/experience/src/routes.ts]: How to differentiate `1.10` and `11.0`?

**Review by tolerants — CHANGES_REQUESTED** [review]

File: libs/platform/experience/src/routes.ts (hunk 1 above)
Comment (tobi-or-not-tobi) [inline libs/platform/experience/src/routes.ts]: what about `1_1_0`?

Comment (tobi-or-not-tobi) [reply libs/platform/experience/src/routes.ts]: https://github.com/spryker/oryx/pull/725#discussion_r1305358958

Comment (dunqan) [reply libs/platform/experience/src/routes.ts]: 👍 we decided to go with `1_1`

Comment (dunqan) [reply libs/platform/experience/src/routes.ts]: we decided to sacrifice support for `1.10` for the time being.

File: apps/storefront/.env.template [hunk 2]
```diff
@@ -30,3 +30,6 @@ ORYX_STORYBLOK_TOKEN=
 # Used to set a preset name, available options: b2b, b2c.
 # Default preset is b2c (storefront).
 ORYX_PRESET=
+
+# Used to set a feature level, available options: latest, 1_0, 1_1, etc.
```
Comment (tobi-or-not-tobi) [inline apps/storefront/.env.template]: can we add that only major and minor are needed?

File: apps/storefront/vite.config.common.ts [hunk 3]
```diff
@@ -1,12 +1,18 @@
...
 
+dotenv.config();
+
 export const viteConfig = {
   index: './src',
   // vite starts paths from index dir
   root: '../',
   envPrefix: ['ORYX', 'FES', 'SCOS', 'STORE'],
+  define: {
+    __ORYX_FEATURE_LEVEL__: `${process.env.ORYX_FEATURE_LEVEL ?? 'latest'}`,
```
Comment (tobi-or-not-tobi) [inline apps/storefront/vite.config.common.ts]: Wonder if latest is a good fallback. It means that if customers do not use this, they will get breaking changes. I'd rather have it more conservative.

File: apps/storefront/.env.template [hunk 4]
```diff
@@ -30,3 +30,6 @@ ORYX_STORYBLOK_TOKEN=
...
+ORYX_FEATURE_LEVEL=
```
Comment (tolerants) [inline apps/storefront/.env.template]: Maybe `ORYX_FEATURE_VERSION`?

**Review by tobi-or-not-tobi — CHANGES_REQUESTED** [review]

File: apps/storefront/.env.template (hunk 4 above)
Comment (tobi-or-not-tobi) [reply apps/storefront/.env.template]: `ORYX_FEATURE_VERSION` doesn't sound right, as this is about opt-in to the latest minor.

File: apps/storefront/vite.config.common.ts (hunk 3 above)
Comment (dunqan) [reply apps/storefront/vite.config.common.ts]: 👍

File: apps/storefront/.env.template (hunk 2 above)
Comment (dunqan) [reply apps/storefront/.env.template]: 👍

**Review by tobi-or-not-tobi — APPROVED** [review]

**Review by tolerants — APPROVED** [review]

**Pull request merged**